src/test/java/io/supertokens/test/ResetPasswordAPITest2_4.java

```diff
@@ -45,4 +50,255 @@ public void beforeEach() {
+            requestBody.addProperty("token", "randomToken");
+            requestBody.addProperty("newPassword", "randomPassword");
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/user/password/reset", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_4ForTests());
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                assertTrue(e.statusCode == 400 &&
+                        e.getMessage()
+                                .equals("Http error. Status Code: 400. Message: Field name 'method' is invalid in " +
+                                        "JSON input"));
+            }
+        }
+
+        {
+            JsonObject requestBody = new JsonObject();
+            requestBody.addProperty("method", "token");
+            requestBody.addProperty("newPassword", "randomPassword");
+
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/user/password/reset", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_4ForTests());
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                assertTrue(e.statusCode == 400 &&
+                        e.getMessage()
+                                .equals("Http error. Status Code: 400. Message: Field name 'token' is invalid in " +
+                                        "JSON input"));
+            }
+        }
+
+        {
+            JsonObject requestBody = new JsonObject();
+            requestBody.addProperty("method", "token");
+            requestBody.addProperty("token", "randomToken");
+            
+            try {
+                io.supertokens.test.httpRequest.HttpRequest
+                        .sendJsonPOSTRequest(process.getProcess(), "",
+                                "http://localhost:3567/recipe/user/password/reset", requestBody, 1000,
+                                1000,
+                                null, Utils.getCdiVersion2_4ForTests());
+                throw new Exception("Should not come here");
+            } catch (io.supertokens.test.httpRequest.HttpResponseException e) {
+                assertTrue(e.statusCode == 400 &&
+                        e.getMessage()
+                                .equals("Http error. Status Code: 400. Message: Field name 'newPassword' is invalid " +
+                                        "in " +
+                                        "JSON input"));
+            }
+        }
+
```

Comment:
You also need to test for when the newPassword is `""`. I think this is mentioned here and in the sign up API.